### PR TITLE
Updates for Guzzle 6 to supercede Guzzle 5.

### PR DESCRIPTION
**Pull request by fideloper**
This creates a Guzzle6Adapter, replacing the Guzzle5Adapter.

The adapter and PHPSpec tests are both updated.

Guzzle 6 shares the same namespace as guzzle 5, and so cannot run side by side guzzle 5.
Guzzle 6 also requires php 5.5 as a new minimum, thus increasing this projects minimum PHP version.

I understand if you don't want to pull this in just yet (or ever!) since it both removes Guzzle 5 (not backwards compatible) and increases the PHP min version to 5.5

I suspect it may be *possible* to keep both Guzzle 5 and Guzzle 6 adapters, but testing and development becomes problematic since both share the same namespace / files. 

You can't have 5 and 6 installed at the same time, so you end up with 2 separate copies. However even then, testing becomes problematic as it needs to decide which version to test, and can't test the whole project all at once.

Cheers!

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,13 +27,13 @@
     },
 
     "require": {
-        "php": ">=5.4.0"
+        "php": ">=5.5.0"
```
Comment (GrahamCampbell) [inline composer.json]: please don't do that

File: composer.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "name"       : "toin0u/digitalocean-v2",
     "type"       : "library",
-    "description": "DigitalOcean API v2 PHP 5.4+ library",
+    "description": "DigitalOcean API v2 PHP 5.5+ library",
```
Comment (GrahamCampbell) [inline composer.json]: please don't do that

**Comment by GrahamCampbell** [comment]
Please keep BOTH a guzzle 5 adapter and a guzzle 6 adapter since having guzzle at all is optional.

**Comment by GrahamCampbell** [comment]
> I suspect it may be possible to keep both Guzzle 5 and Guzzle 6 adapters, but testing and development becomes problematic since both share the same namespace / files.

Yeh, we can have both, but we can't use them both at the same time, but that's fine.

**Comment by fideloper** [comment]
The issue there is that when setting both 5 and 6 in `require-dev`, composer ALWAYS grabs the latest version, so it's actually hard to develop on this package and get guzzle5.

Because Guzzle version 5 and 6 can't coincide on one project, I don't know how to make a passing test that actually covers both. We'd have to  detect the version of Guzzle used and then decide which tests to run.

That might be possible in `function it_is_initializable()` of the Guzzle spec function, since that's the only real difference in terms of the test.

Looks like we can do:

```php
function it_is_initializable()
{
    if (version_compare(GuzzleHttp\Client::version, '6.0.0', '<'))
    {
         $this->shouldHaveType('DigitalOceanV2\Adapter\Guzzle5Adapter');
    } else {
        $this->shouldHaveType('DigitalOceanV2\Adapter\Guzzle6Adapter');
    }
}
```

How's that sound?

**Comment by GrahamCampbell** [comment]
> The issue there is that when setting both 5 and 6 in require-dev, composer ALWAYS grabs the latest version, so it's actually hard to develop on this package and get guzzle5.

It'll get guzzle 5 on php 5.4 though.

**Comment by GrahamCampbell** [comment]
> It looks like we can do something like this

Yeh, that works. ;)

**Comment by fideloper** [comment]
Gotcha, makes sense. I'll get some updates to this probabably later today. :+1:

**Comment by GrahamCampbell** [comment]
Great. Thanks. ;)

**Comment by fideloper** [comment]
I was wrong about the tests, the changes to test guzzle 5 vs 6 are larger. 

Guzzle 5 never had a phpspec test which I didn't realize at first (So guzzle 5 and 6 are both currently not covered under phpspec).

I'll test this out by using it first to shake out anything obvious, and then see if I can get some phpspec action going.

Latest commit brought Guzzle5 support back.

File: composer.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
         "kriswallsmith/buzz": "~0.10",
         "guzzle/guzzle"     : "~3.7",
         "guzzlehttp/guzzle" : "~5.0",
+        "guzzlehttp/guzzle" : "~6.0",
```
Comment (GrahamCampbell) [inline composer.json]: you can't add this dependency twice - json is parsed so one of these ignored

File: composer.json [hunk 4]
```diff
@@ -41,7 +42,8 @@
     "suggest": {
         "kriswallsmith/buzz": "To use BuzzAdapter, require kriswallsmith/buzz:~0.10.",
         "guzzle/guzzle"     : "To use GuzzleAdapter, require guzzle/guzzle:~3.7.",
-        "guzzlehttp/guzzle" : "To use GuzzleV5Adapter, require guzzlehttp/guzzle:~5.0."
+        "guzzlehttp/guzzle" : "To use GuzzleV5Adapter, require guzzlehttp/guzzle:~5.0.",
+        "guzzlehttp/guzzle" : "To use GuzzleV6Adapter, require guzzlehttp/guzzle:~6.0."
```
Comment (GrahamCampbell) [inline composer.json]: again, you can't duplicate the same key

File: composer.json (hunk 3 above)
Comment (GrahamCampbell) [inline composer.json]: it's a bit like pushing to the same key twice in a php array

File: composer.json [hunk 5]
```diff
@@ -34,6 +34,7 @@
         "kriswallsmith/buzz": "~0.10",
         "guzzle/guzzle"     : "~3.7",
         "guzzlehttp/guzzle" : "~5.0",
```
Comment (GrahamCampbell) [inline composer.json]: what we need is `~5.0|~6.0` here

File: src/Adapter/Guzzle5Adapter.php [hunk 6]
```diff
@@ -129,4 +129,4 @@ protected function handleResponse(CompleteEvent $event)
 
         throw new \RuntimeException(sprintf('[%d] %s (%s)', $code, $content->message, $content->id), $code);
     }
-}
+}
```
Comment (GrahamCampbell) [inline src/Adapter/Guzzle5Adapter.php]: please revert that change

File: spec/DigitalOceanV2/Adapter/GuzzleAdapterSpec.php [hunk 7]
```diff
@@ -81,4 +81,4 @@ function it_returns_last_response_header($client, Request $request, Response $re
         $this->getLatestResponseHeaders()->shouldBeArray();
         $this->getLatestResponseHeaders()->shouldHaveCount(3);
     }
-}
+}
```
Comment (GrahamCampbell) [inline spec/DigitalOceanV2/Adapter/GuzzleAdapterSpec.php]: and this one

File: .travis.yml [hunk 8]
```diff
@@ -5,6 +5,7 @@ php:
   - 5.5.9
   - 5.5
   - 5.6
+  - 7.0
```
Comment (GrahamCampbell) [inline .travis.yml]: please revet that change too to keep this PR atomic

Comment (GrahamCampbell) [inline .travis.yml]: infact, i moved it to https://github.com/toin0u/DigitalOceanV2/pull/114

**Comment by fideloper** [comment]
Did some quick testing, looks like the following works:

```json
"require": {
    "guzzlehttp/guzzle": ">=5.0,<7.0"
},
```

Greater than or equal to version 5.0 but less than 7.0. On php 5.5+, this installed v6. On php 5.4, it installed php 5.4 :tada: 

I'll clean these up in a new PR shortly.

**Comment by fideloper** [comment]
Alternatively, we can do:

```
"require": {
    "guzzlehttp/guzzle": "~5.0|~6.0"
},
```

Not sure if you have a preference.

**Pull request closed without merging**